File: versioncontrol/src/main/java/ir/ayantech/versioncontrol/api/VCResponseStatus.java
package ir.ayantech.versioncontrol.api;

import androidx.annotation.Nullable;

import ir.ayantech.versioncontrol.model.VCResponseModel;


/**
 * Created by shadoWalker on 5/9/17.
 */

public interface VCResponseStatus {
    void onSuccess(VersionControlAPI versionControlAPI, String message, @Nullable VCResponseModel responseModel);

    void onFail(VersionControlAPI versionControlAPI,
                String error,
                boolean canTry);
}

File: versioncontrol/src/main/java/ir/ayantech/versioncontrol/api/VCResponseStatus.kt
package ir.ayantech.versioncontrol.api

import ir.ayantech.versioncontrol.model.VCResponseModel

interface VCResponseStatus {
    fun onSuccess(
        versionControlAPI: VersionControlAPI<*, *>?,
        message: String?,
        responseModel: VCResponseModel?
    )

    fun onFail(
        versionControlAPI: VersionControlAPI<*, *>?,
        error: String?,
        canTry: Boolean
    )
}
